sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala

```diff
@@ -48,7 +48,8 @@ case class CreateArray(children: Seq[Expression]) extends Expression {
 
   override def dataType: ArrayType = {
```

Comment:
why `CreateArray` doesn't extend `ComplexTypeMergingExpression`?

Reply:
Because the data type of the `CreateArray` itself is not the merged type but `ArrayType`.